errdefs/defs.go

```diff
@@ -43,6 +43,11 @@ type ErrNotModified interface {
 	NotModified()
 }
 
+// ErrAlreadyExists is a special case of ErrNotModified which signals that the desired object already exists
+type ErrAlreadyExists interface {
```

Comment:
Saw @dnephin already commented on this type https://github.com/moby/moby/pull/35638/files#r160793335, just wanted to add that naming it `ErrAlreadyExists` "feels" like an `ErrConflict`.

`ErrAlreadyExists` and `ErrNotModified` inside the API are converted to a HTTP 304 (`http.StatusNotModified`), which is not an _error_ condition, so perhaps we should have a look at where they're used, and if it's correct that an error is returned at all.

(not for this PR, but we may want to add a TODO or issue somewhere)

Reply:
Indeed we could switch it... it may make since to do this.

Reply:
"Make sense" because it's likely safer to return a conflict than a not modified.